sygnal/helper/proxy/__init__.py

```diff
@@ -0,0 +1,15 @@
+from urllib.parse import urlparse
+
+
+def decompose_http_proxy_url(proxy_url):
+    url = urlparse(proxy_url, scheme="http")
+
+    if not url.hostname:
+        raise RuntimeError("Proxy URL did not contain a hostname! Please specify one.")
+
+    if url.scheme != "http":
+        raise RuntimeError(
+            f"Unknown proxy scheme {url.scheme};" f" only 'http' is supported."
```

Comment:
I'm not familiar with `f""` syntax; how is this different from

```suggestion
            f"Unknown proxy scheme {url.scheme}; only 'http' is supported."
```

?

Reply:
It isn't (nothing to do with the f-ness in this case), unfortunately it often happens that I/my IDE splits multiple lines into e.g.

```python
spqr(
    "This is a long line of text addressed to"
    " Monsieur vdH"
)
```

then if e.g. indentation changes later on, `black` will move them to the same line but not rejoin them together (and I often don't spot them).